src/Rules/Functions/PrintfParameterTypeRule.php

```diff
@@ -100,15 +101,23 @@ public function processNode(Node $node, Scope $scope): array
 			new NullType(),
 		);
 		// Type on the left can go to the type on the right, but not vice versa.
-		$allowedTypeNameMap = [
-			'strict-int' => 'int',
-			'int' => 'castable to int',
-			'float' => 'castable to float',
-			// These are here just for completeness. They won't be used because, these types are already enforced by
-			// CallToFunctionParametersRule.
-			'string' => 'castable to string',
-			'mixed' => 'castable to string',
-		];
+		$allowedTypeNameMap = $this->checkStrictPrintfPlaceholderTypes
+			? [
+				'strict-int' => 'int',
+				'int' => 'int',
+				'float' => 'float',
+				'string' => 'int|float|string|Stringable',
```

Comment:
This seems a bit long and it's not completely accurate either (it accepts classes with `__toString()` even if they don't implement `Stringable`). The other alternative I considered was just `string`, but that might be a bit confusing to users. Is there anything better?

Reply:
Let's try `__stringandstringable`. I can see you're already using `new StringAlwaysAcceptingObjectWithToStringType` in a related place.